assets/js/frontend-components/price-slider/index.js

```diff
@@ -0,0 +1,277 @@
+/**
+ * External dependencies
+ */
+import { sprintf, __ } from '@wordpress/i18n';
+import { Component, createRef } from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+
+const get = ( obj, path, defaultValue ) => {
+	const result = String.prototype.split.call( path, /[,[\].]+?/ )
+		.filter( Boolean )
+		.reduce( ( res, key ) => ( res !== null && res !== undefined ) ? res[ key ] : res, obj );
+	return ( result === undefined || result === obj ) ? defaultValue : result;
+};
+
+class PriceSlider extends Component {
+	constructor( props ) {
+		const { min, max } = props;
+		super( ...arguments );
+		this.state = {
+			currentMin: min,
+			currentMax: max,
+			inputMin: this.formatCurrencyForInput( min ),
+			inputMax: this.formatCurrencyForInput( max ),
+		};
+		this.minInput = createRef();
+		this.maxInput = createRef();
+		this.minRange = createRef();
+		this.maxRange = createRef();
+		this.onDrag = this.onDrag.bind( this );
+		this.onInputChange = this.onInputChange.bind( this );
+		this.onInputBlur = this.onInputBlur.bind( this );
+		this.findClosestRange = this.findClosestRange.bind( this );
+		this.validateValues = this.validateValues.bind( this );
+		this.getProgressStyle = this.getProgressStyle.bind( this );
+		this.formatCurrencyForInput = this.formatCurrencyForInput.bind( this );
+	}
+
+	componentDidUpdate( prevProps, prevState ) {
+		if ( prevState.currentMin !== this.state.currentMin || prevState.currentMax !== this.state.currentMax ) {
+			const { onChange } = this.props;
+
+			onChange( {
+				min: this.state.currentMin,
+				max: this.state.currentMax,
+			} );
+		}
+	}
+
+	formatCurrencyForInput( value ) {
+		if ( '' === value ) {
+			return '';
+		}
+		const formattedNumber = parseInt( value, 10 );
+		const currencySymbol = get( wcSettings, [ 'currency', 'symbol' ], '$' );
+		const priceFormat = get( wcSettings, [ 'currency', 'price_format' ], '%1$s%2$s' );
+
+		if ( '' === formattedNumber ) {
+			return formattedNumber;
+		}
+
+		const formattedValue = sprintf( priceFormat, currencySymbol, formattedNumber );
+		const txt = document.createElement( 'textarea' );
+		txt.innerHTML = formattedValue;
+		return txt.value;
+	}
+
+	validateValues( values, isMin ) {
+		const { min, max, step } = this.props;
+
+		let minValue = parseInt( values[ 0 ], 10 ) || min;
+		let maxValue = parseInt( values[ 1 ], 10 ) || step;
```

Comment:
the `||` would kick in it a 0 was parsed. 

In the case of max, we want to keep it just above the min value, not 0, and definitely not max because the slider would jump back to max if you tried to drag it all the way to the left.